services/motion/builtin/move_request.go

```diff
@@ -214,6 +302,34 @@ func (ms *builtIn) newMoveOnGlobeRequest(
 		}
 	}
 
+	var obstacleDetectors []map[resource.Name]vision.Service
+	for _, obstacleDetectorNamePair := range motionCfg.ObstacleDetectors {
+		pair := make(map[resource.Name]vision.Service)
+		visionServiceName := obstacleDetectorNamePair.VisionServiceName
+		visionSvc, ok := ms.visionServices[visionServiceName]
+		if !ok {
+			return nil, resource.DependencyNotFoundError(visionServiceName)
+		}
+		cameraName := obstacleDetectorNamePair.CameraName
+		// make sure the specified pairs are valid
+		switch {
+		case visionServiceName.ContainsRemoteNames() && cameraName.ContainsRemoteNames():
+			// vision services which are remote only have access to remote cameras
+			pair[cameraName] = visionSvc
+		case visionServiceName.ContainsRemoteNames() && !cameraName.ContainsRemoteNames():
+			// log that a remote vision service cannot use a camera found on the main part of a robot
+			ms.logger.Debugf("%s is remote and %s is not, so they cannot be paired together", visionServiceName, cameraName)
```

Comment:
I think this should just be an error

Reply:
Also as more of a general comment I think it would be frustrating as a user if I misconfigure something and expect it to work but the only indication that I did something wrong is a log message only findable in debug mode.  I don't think we need to be afraid to error in the case someone actually did something wrong